### ISSUE TITLE
If used with Rails, we've got a TypeError: no implicit conversion of String into Integer

### ISSUE THREAD

**Issue by guizmaii** [ms-ati/rumonade#8]
We have the same issue than this thread on StackOverflow when we use this gem with a Rails 3 project : 
http://stackoverflow.com/questions/25844495/typeerror-no-implicit-conversion-of-string-into-integer-on-association

**Comment by ms-ati** [comment]
Hi @guizmaii, thanks for using the gem. I'd be happy to help, but I'm afraid I'll need a bit more context. How are you using Rumonade in the Rails 3 project? Do you have a stack trace that you could share of where the error was raised?

**Comment by guizmaii** [comment]
**How are you using Rumonade in the Rails 3 project?**

Something Like that : 
```ruby
def find_something(searched_thing)
 Option(a_hash.find { |k, v| complex_matching_algo(k, searched_thing) }).flat_map { |k, v| v }
end
```
It's the only thing that I do with ```rumonade```. I search something in a Hash, and when it's found, I get the value.
No Rails in this code, only pure Ruby. 
If I remove ```rumonade``` from my Gemfile, then I ```bundle install``` and I launch my test suite. My tests are green (even if I keep the ```Option``` in  the ```find_something``` method because this method is not already tested)
If I add ```rumonade``` then my tests are red.

**Do you have a stack trace that you could share of where the error was raised?**

```ruby
no implicit conversion of String into Integer (TypeError)
./app/models/model_name.rb:24:in `model_method_that_make_an_eager_query'
...
```
I can't show you more, sorry. 

The problem happens when I eager load a model relation.

To reproduce, it should be easy. 
Create a new Rails 3 project, add ```rumonade``` to ```Gemfile``` and declare two models. 
For example:

```ruby
class User
  has_many :articles

  def self.eager_all
    self.includes(:articles).all
  end
end
```

```ruby
class Article
  belongs_to :user
end
```

Now, somewhere just call the ```eager_all``` method.
It should produce the ```TypeError```

**Comment by ms-ati** [comment]
Excellent, thank you for this help. I will try to reproduce and fix.

**Comment by ms-ati** [comment]
Thank you @guizmaii for excellent steps to reproduce. In branch [issue-8-reproduce](/ms-ati/rumonade/tree/issue-8-reproduce) I have it set up just as you described, and indeed I can reproduce the failure.

My hypothesis is that a change to either `Array` or `Hash` is most likely to be the cause of this -- it's hard to know exactly what magic rails is doing. I'll use the debugger to see what the stack trace can show me.

**Comment by guizmaii** [comment]
We also think that the problem certainly come from the changes made to Hash, as said here: http://stackoverflow.com/a/25845658/2431728

But we can't help more, sorry.

**Comment by ms-ati** [comment]
Hi @guizmaii, 

While I continue to investigate, here's a workaround that may work for you: https://github.com/ms-ati/rumonade/commit/450d31a7e99f3fe534e7d30478f3a7834d2eeb07

Modify your Gemfile like this:

```ruby
## Workaround issue #8 for using Option only
gem 'rumonade', :require => 'rumonade/option'
```

**Comment by ms-ati** [comment]
Ok, I've narrowed the actual problem down to `Array`. Continuing to investigate.

**Comment by ms-ati** [comment]
Ok @guizmaii, this is fixed in v0.4.4

Thanks again for your bug report and clear steps to reproduce.

**Comment by guizmaii** [comment]
Thanks for your quick answers and your quick fix !
I'll test that ASAP.
